onnxscript/function_libs/torch_lib/graph_building.py

```diff
@@ -281,12 +277,38 @@ def eval_function(  # type: ignore[override]
         return self._graph.add_function_call(function, inputs, attributes)
 
 
+def _add_initializers(
+    model_proto: onnx.ModelProto, initializers: Mapping[str, torch.Tensor]
+) -> None:
+    """Add initializers to the model proto."""
+    tensor_protos = []
+
+    for name, tensor in initializers.items():
+        tensor = tensor.detach().cpu().contiguous()
+        # Take the raw data directly from the tensor to avoid the overhead of
+        # data manipulation in onnx.helper.make_tensor
+        raw_data = bytes(
+            (ctypes.c_ubyte * tensor.element_size() * tensor.numel()).from_address(
+                tensor.data_ptr()
+            )
+        )
+        tensor_proto = onnx.helper.make_tensor(
```

Comment:
consider optionally making it `external_data` so you never keep the model in memory.

Reply:
I think that's what we wanted (return a single model proto, so the interface is uniform)? If we didn't need the initializers we can set include_initializers to False. The exporter is responsible for deciding what to do with the model (save to external data, serialize etc.)

Reply:
This way to_model_proto does not have side effects.

Reply:
Are you concerned about the case when we don't have enough memory so we don't want to duplicate the data? I wonder how we want to design the export api for this case?

Reply:
Yes. I meant. TensorProto can be stored as schema (name, shape, element_type) plus a path to a file containing its content -- it's a valid in-memory TensorProto. If we optionally enable it, we can make this part of code scalable to huge scale models. I haven't gone through all parts end-to-end, so this idea might be local instead global necessary.

Reply:
There are multiple fields you can use under TensorProto to store tensor content; `float_data`, `raw_data`, and `external_data`. I feel we should let use to optionally use `external_data`.

Ref: https://github.com/onnx/onnx/blob/ec38b7c4cf3656b9115a68f525017a6e69aa8773/onnx/onnx-ml.proto#L608

Reply:
I see! That makes sense. Let me think more on this